staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy_test.go

```diff
@@ -655,21 +659,509 @@ func TestDropDisabledFields(t *testing.T) {
+									OptionalOldSelf: &tru,
+								},
+							},
+							Dependencies: apiextensions.JSONSchemaDependencies{
+								"test": apiextensions.JSONSchemaPropsOrStringArray{
+									Schema: &apiextensions.JSONSchemaProps{
+										Type: "object",
+										XValidations: apiextensions.ValidationRules{
+											{
+												Rule:            "size(self) > 0",
+												Message:         "size of scoped field should be greater than 0.",
+												OptionalOldSelf: &tru,
+											},
+										},
+									},
+								},
+							},
+							Properties: map[string]apiextensions.JSONSchemaProps{
+								"subRule": {
+									Type: "object",
+									XValidations: apiextensions.ValidationRules{
+										{
+											Rule:            "isTest == true",
+											Message:         "isTest should be true.",
+											OptionalOldSelf: &tru,
+										},
+									},
+									Properties: map[string]apiextensions.JSONSchemaProps{
+										"isTest": {
+											Type: "boolean",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			oldCRD: nil,
+			expectedCRD: &apiextensions.CustomResourceDefinition{
+				ObjectMeta: metav1.ObjectMeta{Name: "foos.sigs.k8s.io", Annotations: map[string]string{v1beta1.KubeAPIApprovedAnnotation: "valid"}, ResourceVersion: "1"},
+				Spec: apiextensions.CustomResourceDefinitionSpec{
+					Validation: &apiextensions.CustomResourceValidation{
+						OpenAPIV3Schema: &apiextensions.JSONSchemaProps{
+							Type: "object",
+							XValidations: apiextensions.ValidationRules{
+								{
+									Rule:            "size(self) > 0",
+									Message:         "openAPIV3Schema should contain more than 0 element.",
+									OptionalOldSelf: &tru,
+								},
+							},
+							Dependencies: apiextensions.JSONSchemaDependencies{
+								"test": apiextensions.JSONSchemaPropsOrStringArray{
+									Schema: &apiextensions.JSONSchemaProps{
+										Type: "object",
+										XValidations: apiextensions.ValidationRules{
+											{
+												Rule:            "size(self) > 0",
+												Message:         "size of scoped field should be greater than 0.",
+												OptionalOldSelf: &tru,
+											},
+										},
+									},
+								},
+							},
+							Properties: map[string]apiextensions.JSONSchemaProps{
+								"subRule": {
+									Type: "object",
+									XValidations: apiextensions.ValidationRules{
+										{
+											Rule:            "isTest == true",
+											Message:         "isTest should be true.",
+											OptionalOldSelf: &tru,
+										},
+									},
+									Properties: map[string]apiextensions.JSONSchemaProps{
+										"isTest": {
+											Type: "boolean",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:               "Ratcheting, For update, FG disabled, oldCRD OptionalOldSelf in use, don't drop OptionalOldSelfs",
```

Comment:
Do we have a test for `Ratcheting, For update, FG disabled, oldCRD OptionalOldSelf in use, but the value is different than the new OptionalOldSelf, don't drop OptionalOldSelfs`?

Reply:
I had considered the cases which introduiced a new usage which previously did not exist to excercise that case. To be extra explicit, added a new explicit test for that.